modules/services/mpd.nix

```diff
@@ -87,7 +96,14 @@ in {
         '';
       };
 
-     network = {
+      network = {
+         socketActivation = mkOption {
```

Comment:
Could use a boolean option `service.mpd.startWhenNeeded`, like in the NixOS module.

I don't know if it's worth supporting unix sockets when binding to localhost will work just as well. But if you want to keep the unix socket then perhaps rename this option `listenSocket` and remove the `none` alternative.

Reply:
> Could use a boolean option service.mpd.startWhenNeeded, like in the NixOS module

Sure, I'll try to keep the naming consistent.

> I don't know if it's worth supporting unix sockets when binding to localhost will work just as well.

There are some clients where this matters, for example [ncmpcpp](https://github.com/ncmpcpp/ncmpcpp/blob/81cb7a4f85287ce2671beca9023048f4de41ddf0/src/screens/browser.cpp#L524). But if the protocol doesn't care maybe that could be considered a client bug?

Reply:
No worries. It'll be fine to keep the unix socket and perhaps also always create it, I can't imagine a situation where it would hurt having it around. Indeed, I wonder if it wouldn't make sense to simply always use socket activation? Then no new option would be necessary, just to add a news entry and release note informing people.